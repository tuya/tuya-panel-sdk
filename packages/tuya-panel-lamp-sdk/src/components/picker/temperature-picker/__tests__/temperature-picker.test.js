import React from 'react';
import { shallow } from 'enzyme';
import TemperaturePicker from '../index';

describe('TemperaturePicker components', () => {
  const origConsole = console.error;
  beforeEach(() => {
    console.error = () => {};
  });
  afterEach(() => {
    console.error = origConsole;
  });

  it('basic render', () => {
    let value = 1000;
    const wrapper = shallow(<TemperaturePicker value={value} />);
    expect(wrapper).toMatchSnapshot();

    value = 500;
    wrapper.setProps({ value });
  });

  it('componentWillReceiveProps render', () => {
    let disabled = false;
    let hideThumb = false;

    const wrapper = shallow(<TemperaturePicker disabled={disabled} hideThumb={hideThumb} />);
    expect(wrapper).toMatchSnapshot();

    disabled = true;
    wrapper.setProps({ disabled });

    hideThumb = true;
    wrapper.setProps({ hideThumb });

    wrapper.setProps({
      renderThumb: () => {
        return <View />;
      },
    });

    wrapper.unmount();
  });

  it('render touch', () => {
    const wrapper = shallow(<TemperaturePicker disabled={false} value={1000} innerRadius={0} />);

    const pander = wrapper.findWhere(
      c => c.name() === 'View' && !!c.prop('onStartShouldSetResponder') === true
    );

    pander.simulate('startShouldSetResponder', { nativeEvent: { locationX: 45, locationY: 3 } });

    pander.simulate('moveShouldSetResponder');
    pander.simulate('responderGrant', {
      nativeEvent: { touches: [] },
      touchHistory: { touchBank: [] },
    });
    pander.simulate('responderMove', {
      nativeEvent: { touches: [] },
      touchHistory: { touchBank: [] },
    });
    pander.simulate('responderRelease', {
      nativeEvent: { touches: [] },
      touchHistory: { touchBank: [] },
    });

    pander.simulate('responderTerminationRequest');
    expect(wrapper).toMatchSnapshot();
  });

  it('render touch disabled', () => {
    const wrapper = shallow(<TemperaturePicker value={1000} />);

    const pander = wrapper.findWhere(
      c => c.name() === 'View' && !!c.prop('onStartShouldSetResponder') === true
    );
    wrapper.setProps({ disabled: true });
    pander.simulate('startShouldSetResponder', { nativeEvent: { locationX: 45, locationY: 3 } });
    expect(wrapper).toMatchSnapshot();
  });
});
