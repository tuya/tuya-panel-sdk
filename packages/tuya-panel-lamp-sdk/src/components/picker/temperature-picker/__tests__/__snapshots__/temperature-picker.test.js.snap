// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperaturePicker components basic render 1`] = `
<View
  disalbedThumbOpacity={0.4}
  hideThumb={false}
  max={1000}
  min={0}
  offsetAngle={180}
  onChange={[Function]}
  onGrant={[Function]}
  onMove={[Function]}
  onPress={[Function]}
  onRelease={[Function]}
  opacityAnimationDuration={150}
  opacityAnimationValue={1}
  sideOffset={10}
  step={1}
  storageKey="temperature_picker"
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 604,
        "justifyContent": "center",
        "width": 604,
      },
      undefined,
    ]
  }
  touchOffset={0}
  touchThumbRadius={0}
  value={1000}
>
  <AnimatedComponent
    style={
      Object {
        "height": 540,
        "opacity": 1,
        "width": 540,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 540,
          "justifyContent": "center",
          "width": 540,
        }
      }
    >
      <Svg
        height={540}
        preserveAspectRatio="xMidYMid meet"
        width={540}
      >
        <Defs>
          <LinearGradient
            id="temperaturePicker_540138"
            x1="0%"
            x2="100%"
            y1="50%"
            y2="50%"
          >
            <Stop
              key="0"
              offset="0%"
              stopColor="#FFCA5C"
              stopOpacity={1}
            />
            <Stop
              key="1"
              offset="50%"
              stopColor="#fff"
              stopOpacity={1}
            />
            <Stop
              key="2"
              offset="100%"
              stopColor="#CDECFE"
              stopOpacity={1}
            />
          </LinearGradient>
        </Defs>
        <Path
          d="M0 270A270 270 0 1 0 540 270A270 270 0 1 0 0 270Z
    M138 270A132 132 0 1 1 402 270A132 132 0 1 1 138 270Z"
          fill="url(#temperaturePicker_540138)"
        />
      </Svg>
    </View>
  </AnimatedComponent>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 604,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 604,
      }
    }
  />
  <View
    accessibilityLabel="TemperaturePicker"
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    pointerEvents="box-only"
    style={
      Object {
        "height": 604,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 604,
      }
    }
  />
</View>
`;

exports[`TemperaturePicker components componentWillReceiveProps render 1`] = `
<View
  disalbedThumbOpacity={0.4}
  hideThumb={false}
  max={1000}
  min={0}
  offsetAngle={180}
  onChange={[Function]}
  onGrant={[Function]}
  onMove={[Function]}
  onPress={[Function]}
  onRelease={[Function]}
  opacityAnimationDuration={150}
  opacityAnimationValue={1}
  sideOffset={10}
  step={1}
  storageKey="temperature_picker"
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 604,
        "justifyContent": "center",
        "width": 604,
      },
      undefined,
    ]
  }
  touchOffset={0}
  touchThumbRadius={0}
  value={0}
>
  <AnimatedComponent
    style={
      Object {
        "height": 540,
        "opacity": 1,
        "width": 540,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 540,
          "justifyContent": "center",
          "width": 540,
        }
      }
    >
      <Svg
        height={540}
        preserveAspectRatio="xMidYMid meet"
        width={540}
      >
        <Defs>
          <LinearGradient
            id="temperaturePicker_540138"
            x1="0%"
            x2="100%"
            y1="50%"
            y2="50%"
          >
            <Stop
              key="0"
              offset="0%"
              stopColor="#FFCA5C"
              stopOpacity={1}
            />
            <Stop
              key="1"
              offset="50%"
              stopColor="#fff"
              stopOpacity={1}
            />
            <Stop
              key="2"
              offset="100%"
              stopColor="#CDECFE"
              stopOpacity={1}
            />
          </LinearGradient>
        </Defs>
        <Path
          d="M0 270A270 270 0 1 0 540 270A270 270 0 1 0 0 270Z
    M138 270A132 132 0 1 1 402 270A132 132 0 1 1 138 270Z"
          fill="url(#temperaturePicker_540138)"
        />
      </Svg>
    </View>
  </AnimatedComponent>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 604,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 604,
      }
    }
  />
  <View
    accessibilityLabel="TemperaturePicker"
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    pointerEvents="box-only"
    style={
      Object {
        "height": 604,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 604,
      }
    }
  />
</View>
`;

exports[`TemperaturePicker components render touch 1`] = `
<View
  disalbedThumbOpacity={0.4}
  hideThumb={false}
  max={1000}
  min={0}
  offsetAngle={180}
  onChange={[Function]}
  onGrant={[Function]}
  onMove={[Function]}
  onPress={[Function]}
  onRelease={[Function]}
  opacityAnimationDuration={150}
  opacityAnimationValue={1}
  sideOffset={10}
  step={1}
  storageKey="temperature_picker"
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 604,
        "justifyContent": "center",
        "width": 604,
      },
      undefined,
    ]
  }
  touchOffset={0}
  touchThumbRadius={0}
  value={1000}
>
  <AnimatedComponent
    style={
      Object {
        "height": 540,
        "opacity": 1,
        "width": 540,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 540,
          "justifyContent": "center",
          "width": 540,
        }
      }
    >
      <Svg
        height={540}
        preserveAspectRatio="xMidYMid meet"
        width={540}
      >
        <Defs>
          <LinearGradient
            id="temperaturePicker_540270"
            x1="0%"
            x2="100%"
            y1="50%"
            y2="50%"
          >
            <Stop
              key="0"
              offset="0%"
              stopColor="#FFCA5C"
              stopOpacity={1}
            />
            <Stop
              key="1"
              offset="50%"
              stopColor="#fff"
              stopOpacity={1}
            />
            <Stop
              key="2"
              offset="100%"
              stopColor="#CDECFE"
              stopOpacity={1}
            />
          </LinearGradient>
        </Defs>
        <Path
          d="M0 270A270 270 0 1 0 540 270A270 270 0 1 0 0 270Z
    M270 270A0 0 0 1 1 270 270A0 0 0 1 1 270 270Z"
          fill="url(#temperaturePicker_540270)"
        />
      </Svg>
    </View>
  </AnimatedComponent>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 604,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 604,
      }
    }
  />
  <View
    accessibilityLabel="TemperaturePicker"
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    pointerEvents="box-only"
    style={
      Object {
        "height": 604,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 604,
      }
    }
  />
</View>
`;

exports[`TemperaturePicker components render touch disabled 1`] = `
<View
  disalbedThumbOpacity={0.4}
  hideThumb={false}
  max={1000}
  min={0}
  offsetAngle={180}
  onChange={[Function]}
  onGrant={[Function]}
  onMove={[Function]}
  onPress={[Function]}
  onRelease={[Function]}
  opacityAnimationDuration={150}
  opacityAnimationValue={1}
  sideOffset={10}
  step={1}
  storageKey="temperature_picker"
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 604,
        "justifyContent": "center",
        "width": 604,
      },
      undefined,
    ]
  }
  touchOffset={0}
  touchThumbRadius={0}
  value={1000}
>
  <AnimatedComponent
    style={
      Object {
        "height": 540,
        "opacity": 1,
        "width": 540,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 540,
          "justifyContent": "center",
          "width": 540,
        }
      }
    >
      <Svg
        height={540}
        preserveAspectRatio="xMidYMid meet"
        width={540}
      >
        <Defs>
          <LinearGradient
            id="temperaturePicker_540138"
            x1="0%"
            x2="100%"
            y1="50%"
            y2="50%"
          >
            <Stop
              key="0"
              offset="0%"
              stopColor="#FFCA5C"
              stopOpacity={1}
            />
            <Stop
              key="1"
              offset="50%"
              stopColor="#fff"
              stopOpacity={1}
            />
            <Stop
              key="2"
              offset="100%"
              stopColor="#CDECFE"
              stopOpacity={1}
            />
          </LinearGradient>
        </Defs>
        <Path
          d="M0 270A270 270 0 1 0 540 270A270 270 0 1 0 0 270Z
    M138 270A132 132 0 1 1 402 270A132 132 0 1 1 138 270Z"
          fill="url(#temperaturePicker_540138)"
        />
      </Svg>
    </View>
  </AnimatedComponent>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 604,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 604,
      }
    }
  />
  <View
    accessibilityLabel="TemperaturePicker"
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    pointerEvents="box-only"
    style={
      Object {
        "height": 604,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 604,
      }
    }
  />
</View>
`;
